hazelcast-hibernate5/src/main/java/com/hazelcast/hibernate/serialization/Value.java

```diff
@@ -94,4 +95,23 @@ public int getClassId() {
         return HibernateDataSerializerHook.VALUE;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        if (!super.equals(o)) {
+            return false;
+        }
+        Value value1 = (Value) o;
+        return timestamp == value1.timestamp && Objects.equals(value, value1.value);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), timestamp, value);
```

Comment:
Could it be something like below (not complete, just for the demonstration):
```
int result = super.hashCode();
result = 31 * result + (value == null ? 0 : value.hashCode());
result = 31 * result + Long.hashCode(timestamp);
return result;
```
 to prevent creating an intermediate array (as parameter list) and boxing/unboxing timestamp during Objects#hash - as we have only 2 fields here?

Reply:
Good point, I will use a different template then (yes, I did not code it myself)